### PR TITLE
Initial commit (#41)

### PR DESCRIPTION
**Pull request by nylchr**
Get List of Agile Boards
Get Agile Configuration by ID
Get Sprint by ID

### CODE REVIEW

File: src/YouTrackSharp/AgileBoards/AgileSettings.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        public string Query { get; set; }
+
+        [JsonProperty("columnSettings")]
+        public ColumnSettings ColumnSettings { get; set; }
+
+        [JsonProperty("timeField")]
+        public Field TimeField { get; set; }
+
+        [JsonProperty("projectsColors")]
+        public object ProjectsColors { get; set; }
```
Comment (maartenba) [inline src/YouTrackSharp/AgileBoards/AgileSettings.cs]: nit: rename the C# property to `ProjectColors`

File: src/YouTrackSharp/AgileBoards/AgileSettings.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+        [JsonProperty("colorConfig")]
+        public ColorConfig ColorConfig { get; set; }
+
+        [JsonProperty("completeBacklogHierarhy")]
+        public bool CompleteBacklogHierarhy { get; set; }
```
Comment (maartenba) [inline src/YouTrackSharp/AgileBoards/AgileSettings.cs]: What does this property provide? Maybe add XML doc comments.

File: src/YouTrackSharp/AgileBoards/AgileBoardService.cs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        }
+
+        /// <summary>
+        /// Get the agile board with the specified id.
+        /// </summary>
+        /// <remarks>Uses the REST API <a href="https://www.jetbrains.com/help/youtrack/standalone/Get-Agile-Configuration-by-ID.html">Get Agile Configuration by ID</a>.</remarks>
+        ///  <param name="agileId">Id of the agile board containing the sprint.</param>
+        /// <returns>An <see cref="AgileSettings" /> that match the specified parameter.</returns>
+        /// <exception cref="T:System.Net.HttpRequestException">When the call to the remote YouTrack server instance failed.</exception>
+        public async Task<AgileSettings> GetAgileBoard(string agileId)
```
Comment (maartenba) [inline src/YouTrackSharp/AgileBoards/AgileBoardService.cs]: Can we rename `agileId` to `agileBoardId` ?

File: src/YouTrackSharp/AgileBoards/AgileBoardService.cs [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+        /// <summary>
+        /// Get sprint by id
+        /// </summary>
+        /// <remarks>Uses the REST API <a href="https://www.jetbrains.com/help/youtrack/standalone/Get-Sprint-by-ID.html">Get Sprint by ID</a>.</remarks>
+        ///  <param name="agileId">Id of the agile board containing the sprint.</param>
+        ///  <param name="sprintId">Id of the sprint.</param>
+        /// <returns>A <see cref="Sprint" /> that match the specified parameters.</returns>
+        /// <exception cref="T:System.Net.HttpRequestException">When the call to the remote YouTrack server instance failed.</exception>
+        public async Task<Sprint> GetSprint(string agileId, string sprintId)
```
Comment (maartenba) [inline src/YouTrackSharp/AgileBoards/AgileBoardService.cs]: Can we rename `agileId` to `agileBoardId` ?

File: src/YouTrackSharp/AgileBoards/AgileSettings.cs [hunk 5]
```diff
@@ -0,0 +1,44 @@
+﻿using Newtonsoft.Json;
+using System.Collections.Generic;
+
+namespace YouTrackSharp.AgileBoards
+{
+    public class AgileSettings
```
Comment (maartenba) [inline src/YouTrackSharp/AgileBoards/AgileSettings.cs]: Add XML doc comments to all properties.

File: src/YouTrackSharp/AgileBoards/Backlog.cs [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+        public string Value { get; set; }
+
+        [JsonProperty("name")]
+        public string Name { get; set; }
+
+        [JsonProperty("visibleForGroup")]
+        public string VisibleForGroup { get; set; }
+
+        [JsonProperty("updatableByGroup")]
+        public object UpdatableByGroup { get; set; }
```
Comment (maartenba) [inline src/YouTrackSharp/AgileBoards/Backlog.cs]: Is `object` the most specific type here? Can this be made more specific?

File: src/YouTrackSharp/AgileBoards/Backlog.cs [hunk 7]
```diff
@@ -0,0 +1,19 @@
+﻿using Newtonsoft.Json;
+
+namespace YouTrackSharp.AgileBoards
+{
+    public class Backlog
+    {
+        [JsonProperty("value")]
```
Comment (maartenba) [inline src/YouTrackSharp/AgileBoards/Backlog.cs]: Add XML docs to all properties

File: src/YouTrackSharp/AgileBoards/ColumnSettings.cs [hunk 8]
```diff
@@ -0,0 +1,14 @@
+﻿using Newtonsoft.Json;
+using System.Collections.Generic;
+
+namespace YouTrackSharp.AgileBoards
+{
+    public class ColumnSettings
+    {
+        [JsonProperty("field")]
```
Comment (maartenba) [inline src/YouTrackSharp/AgileBoards/ColumnSettings.cs]: Add XML docs to all properties

File: src/YouTrackSharp/AgileBoards/ColumnSettings.cs [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+        public Field Field { get; set; }
+
+        [JsonProperty("visibleValues")]
+        public ICollection<VisibleValue> VisibleValues { get; set; }
```
Comment (maartenba) [inline src/YouTrackSharp/AgileBoards/ColumnSettings.cs]: Initialize in constructor

File: src/YouTrackSharp/AgileBoards/AgileSettings.cs [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+
+        [JsonProperty("colorConfig")]
+        public ColorConfig ColorConfig { get; set; }
+
+        [JsonProperty("completeBacklogHierarhy")]
+        public bool CompleteBacklogHierarhy { get; set; }
+
+        public SwimlaneSettings SwimlaneSettings { get; set; }
+        public Backlog Backlog { get; set; }
+        public ICollection<Sprint> Sprints { get; set; }
```
Comment (maartenba) [inline src/YouTrackSharp/AgileBoards/AgileSettings.cs]: Initialize collection in constructor

File: src/YouTrackSharp/AgileBoards/Field.cs [hunk 11]
```diff
@@ -0,0 +1,13 @@
+﻿using Newtonsoft.Json;
+
+namespace YouTrackSharp.AgileBoards
+{
+    public class Field
```
Comment (maartenba) [inline src/YouTrackSharp/AgileBoards/Field.cs]: Add XML doc comments

File: src/YouTrackSharp/AgileBoards/Project.cs [hunk 12]
```diff
@@ -0,0 +1,10 @@
+﻿using Newtonsoft.Json;
+
+namespace YouTrackSharp.AgileBoards
```
Comment (maartenba) [inline src/YouTrackSharp/AgileBoards/Project.cs]: XML doc comments. Also is `id` the only property?

File: src/YouTrackSharp/AgileBoards/Sprint.cs [hunk 13]
```diff
@@ -0,0 +1,23 @@
+﻿using Newtonsoft.Json;
+using System;
+using YouTrackSharp.Json;
+
+namespace YouTrackSharp.AgileBoards
+{
+    public class Sprint
```
Comment (maartenba) [inline src/YouTrackSharp/AgileBoards/Sprint.cs]: XML doc comments

File: src/YouTrackSharp/AgileBoards/Sprint.cs [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+    {
+        [JsonProperty("id")]
+        public string Id { get; set; }
+
+        [JsonProperty("version")]
+        public string Version { get; set; }
+
+        [JsonConverter(typeof(UnixDateTimeOffsetConverter))]
+        [JsonProperty("start")]
+        public DateTime? Start { get; set; }
```
Comment (maartenba) [inline src/YouTrackSharp/AgileBoards/Sprint.cs]: Make the type `DateTimeOffset` if possible?

File: src/YouTrackSharp/AgileBoards/Sprint.cs [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+
+        [JsonConverter(typeof(UnixDateTimeOffsetConverter))]
+        [JsonProperty("finish")]
+        public DateTime? Finish { get; set; }
```
Comment (maartenba) [inline src/YouTrackSharp/AgileBoards/Sprint.cs]: Make the type `DateTimeOffset` if possible?

File: src/YouTrackSharp/AgileBoards/SwimlaneSettings.cs [hunk 16]
```diff
@@ -0,0 +1,17 @@
+﻿using Newtonsoft.Json;
+using System.Collections.Generic;
+
+namespace YouTrackSharp.AgileBoards
+{
+    public class SwimlaneSettings
+    {
+        [JsonProperty("field")]
```
Comment (maartenba) [inline src/YouTrackSharp/AgileBoards/SwimlaneSettings.cs]: Add XML docs and initialize `Values` collection.

File: src/YouTrackSharp/AgileBoards/VisibleValues.cs [hunk 17]
```diff
@@ -0,0 +1,16 @@
+﻿using Newtonsoft.Json;
+
+namespace YouTrackSharp.AgileBoards
+{
+    public class VisibleValue
+    {
```
Comment (maartenba) [inline src/YouTrackSharp/AgileBoards/VisibleValues.cs]: XML docs

File: src/YouTrackSharp/AgileBoards/VisibleValues.cs [hunk 18]
```diff
@@ -0,0 +1,16 @@
...
+        [JsonProperty("value")]
+        public string Value { get; set; }
+
+        [JsonProperty("min")]
+        public object Min { get; set; }
```
Comment (maartenba) [inline src/YouTrackSharp/AgileBoards/VisibleValues.cs]: Can we be more specific than `object`?

File: src/YouTrackSharp/ConnectionExtensions.cs [hunk 19]
```diff
@@ -69,5 +70,15 @@ public static ProjectCustomFieldsService ProjectCustomFieldsService(this Connect
         {
             return new ProjectCustomFieldsService(connection);
         }
+
+        /// <summary>
+        /// Creates a <see cref="AgileBoardService" />.
+        /// </summary>
+        /// <param name="connection">The <see cref="Connection" /> to create a service with.</param>
+        /// <returns><see cref="AgileBoardService" /> for working with YouTrack agile boards.</returns>
+        public static AgileBoardService CreateAgileBoardsService(this Connection connection)
```
Comment (maartenba) [inline src/YouTrackSharp/ConnectionExtensions.cs]: Nit: `CreateAgileBoardService`

File: src/YouTrackSharp/AgileBoards/AgileSettings.cs (hunk 2 above)
Comment (nylchr) [reply src/YouTrackSharp/AgileBoards/AgileSettings.cs]: I have no idea what the property means :) The comment I added states that so someone who knows should probably change that or explain it to me and I'll change it.

File: src/YouTrackSharp/AgileBoards/AgileSettings.cs (hunk 1 above)
Comment (nylchr) [reply src/YouTrackSharp/AgileBoards/AgileSettings.cs]: This seems to be an error in the API. As far as I can tell the property is not included in the xml return, only in json. I removed the property from the class untill further notice.

File: src/YouTrackSharp/AgileBoards/Backlog.cs (hunk 6 above)
Comment (nylchr) [reply src/YouTrackSharp/AgileBoards/Backlog.cs]: Did some research and changed it to string

File: src/YouTrackSharp/AgileBoards/Project.cs (hunk 12 above)
Comment (nylchr) [reply src/YouTrackSharp/AgileBoards/Project.cs]: Yes. To get more data on a project you need to use the https://www.jetbrains.com/help/youtrack/standalone/Get-Accessible-Projects.html API.

**Comment by maartenba** [comment]
Thanks @nylchr ! Merged to dev.

**Pull request merged**